modules/custom/d_p/d_p.module

```diff
@@ -237,9 +247,14 @@ function d_p_preprocess_field(&$variables) {
   // Alter field_d_main_title.
   if ($variables['element']['#field_name'] == 'field_d_main_title') {
     $entity = $variables["element"]["#object"];
-    if ($entity->getEntityTypeId() == 'paragraph') {
-      $helper = new ParagraphModifiersHelper($entity);
-      $variables['heading_tag'] = $helper->getModifier(SettingsWidget::HEADING_TYPE_SETTING_NAME);
+    if ($entity->getEntityTypeId() === 'paragraph') {
+      try {
+        /** @var \Drupal\d_p\Plugin\Field\ConfigurationStorageFieldItemListInterface $setting_field */
+        $setting_field = ConfigurationStorage::getSettingsFieldFromEntity($entity);
+        $variables['heading_tag'] = $setting_field->getSettingValue(ParagraphSettingTypesInterface::HEADING_TYPE_SETTING_NAME);
+      }
+      catch (MissingConfigurationStorageFieldException $exception) {
```

Comment:
Missing logger

Reply:
It's a silent catch. As you can see it is checking only if the given entity is a paragraph type or not. We may not want to put so many error logs for the bundles that do not have/do not support a setting field.